modules/core/src/v2/coins/dot.ts

```diff
@@ -204,15 +186,17 @@ export class Dot extends BaseCoin {
    * @returns {Promise<SignedTransaction>}
    */
   async signTransaction(params: SignTransactionOptions): Promise<SignedTransaction> {
-    const { txHex, signer, prv } = this.verifySignTransactionParams(params);
+    const { txHex, prv } = this.verifySignTransactionParams(params);
     const factory = accountLib.register(this.getChain(), accountLib.Dot.TransactionBuilderFactory);
     const txBuilder = factory.from(txHex);
+    const keyPair = new accountLib.Dot.KeyPair({ prv: prv });
+
     txBuilder
-      .validity(params.txPrebuild.validity)
-      .referenceBlock(params.txPrebuild.referenceBlock)
-      .version(params.txPrebuild.version)
-      .sender({ address: signer })
-      .sign({ key: prv });
+      .validity({ firstValid: params.txPrebuild.transaction.blockNumber })
```

Comment:
nit: we can deconstruct the object `params.txPrebuild.transaction` here to avoid dup